dist/ethereum.js

```diff
@@ -2063,14 +2062,19 @@ var filter = function(options, implementation, formatter) {
         });
     };
 
-    implementation.startPolling(filterId, onMessages, implementation.uninstallFilter);
 
     var watch = function(callback) {
+        implementation.startPolling(filterId, onMessages, implementation.uninstallFilter);
         callbacks.push(callback);
     };
 
     var stopWatching = function() {
         implementation.stopPolling(filterId);
+        callbacks = [];
+    };
+
+    var uninstall = function() {
+        implementation.stopPolling(filterId);
```

Comment:
Stop watching just stops sending FilterChanges requests, and you can start watching agiain. uninstall uninstalls the filter completely.